src/Commands/ToolsCommandGroup.cs

```diff
@@ -431,31 +484,42 @@ private Task<Result> ShowGuildInfoAsync(IUser bot, IGuild guild, CancellationTok
         [Description("Offset from current time")] [Option("offset")]
         string? stringOffset = null)
     {
+        _profiler.Push("timestamp_command");
+        _profiler.Push("preparation");
         if (!_context.TryGetContextIDs(out var guildId, out _, out var executorId))
         {
-            return new ArgumentInvalidError(nameof(_context), "Unable to retrieve necessary IDs from command context");
+            return _profiler.ReportWithResult(new ArgumentInvalidError(nameof(_context),
+                "Unable to retrieve necessary IDs from command context"));
         }
 
+        _profiler.Push("current_user_get");
         var botResult = await _userApi.GetCurrentUserAsync(CancellationToken);
         if (!botResult.IsDefined(out var bot))
         {
             return Result.FromError(botResult);
         }
 
+        _profiler.Pop();
+        _profiler.Push("executor_get");
         var executorResult = await _userApi.GetUserAsync(executorId, CancellationToken);
         if (!executorResult.IsDefined(out var executor))
         {
             return Result.FromError(executorResult);
         }
 
+        _profiler.Pop();
+        _profiler.Push("guild_settings_get");
         var data = await _guildData.GetData(guildId, CancellationToken);
         Messages.Culture = GuildSettings.Language.Get(data.Settings);
 
+        _profiler.Pop();
         if (stringOffset is null)
         {
-            return await SendTimestampAsync(null, executor, CancellationToken);
+            _profiler.Pop();
+            return _profiler.ReportWithResult(await SendTimestampAsync(null, executor, CancellationToken));
```

Comment:
`ReportWithResult` pops all running profilers automagically
```suggestion
            return _profiler.ReportWithResult(await SendTimestampAsync(null, executor, CancellationToken));
```